onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,201 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least three inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 3)) {
+    fail_shape_inference("Col2Im requires 3 inputs");
```

Comment:
this is not right... 

some conditions are unmet because the data is not available at the time of inference and you should not throw in those cases... 

What I meant in my earlier comment is - if something is invalid then you should throw.

hasNInputShapes(ctx, 3) is allowed to be false... it is possible that output_shape is not initializer or constant and therefore you dont know the shape. In this case you should simply return.